knowledge_graphs/query_pagination.py

```
@@ -0,0 +1,382 @@
+        for i, line in enumerate(lines):
+            line_upper = line.strip().upper()
+            if line_upper.startswith('RETURN'):
+                # Replace with count
+                lines[i] = 'RETURN COUNT(*) as count'
+                break
+        
+        # Remove SKIP and LIMIT clauses
+        filtered_lines = []
+        for line in lines:
+            line_upper = line.strip().upper()
+            if not (line_upper.startswith('SKIP') or line_upper.startswith('LIMIT')):
```

Ruff (0.12.2)

Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)